tests/cbmc/proofs/s2n_strcpy/s2n_strcpy_harness.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "utils/s2n_str.h"
+
+#include <assert.h>
+#include <string.h>
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <sys/param.h>
+
+void s2n_strcpy_harness() {
+
+    char *str = ensure_c_str_is_allocated(MAX_STRING_LEN);
+    uint32_t slen = strlen(str);
+    uint32_t buflen;
+    uint32_t index;
+    __CPROVER_assume(buflen < MAX_STRING_LEN);
+    __CPROVER_assume(index < strlen(str));
+    char buf[buflen];
+    char *last;
+    char str_byte = str[index];
+    char *result;
+
+    /* Non-deterministically set initialized (in s2n_mem) to true. */
+    if (nondet_bool()) {
+        s2n_mem_init();
+    }
+
+    /* Non-deterministically set str to NULL. */
+    if(nondet_bool()) {
+        str = NULL;
+    }
+
+
+/* CBMC pointer checks need to be disabled to compare buf and last for
+ * the case where they are the same. */
+#pragma CPROVER check push
```

Comment:
It seems to be a requirement of the code that `last` be a pointer into `buf`. Maybe just do it like this?

```
size_t last_offset;
__CPROVER_ASSUME(last_offset < buflen);
char* last = &buf[last_offset];
```

Reply:
I'll do that, it's much cleaner.